### PR TITLE
drtprod: yaml config support

### LINKED ISSUE #125381: drtprod: implement drt prod in golang rather than shell for better meintenance
Currently, drtprod is a shell script as scripts/drtprod. The script has the logic to do lot of different operations. As we keep adding more commands to the same script, it will grow and will become unmanageable. So, we can move the code to be a golang code similar to roachprod and make it more manageable and navigable.

Jira issue: CRDB-39422

### PR DESCRIPTION
**Pull request by nameisbhaskar**
This implementation of drtprod enables you to define the series of commands in a YAML file and execute the same. This way you can define a sequence of commands for a cluster.

You can find more details in https://cockroachlabs.atlassian.net/l/cp/0nxKZuyP.

With this the expectation is to decommission drtprod script and move all configurations to smaller scripts and yaml.

Fixes: #125381
Epic: None

### CODE REVIEW

**Review by sambhav-jain-16 — APPROVED** [review]
Thanks @nameisbhaskar , this tool is going to be really useful.

One minor comment.
Apart from it LGTM !

File: pkg/cmd/drtprod/cli/handlers.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+// Initialize sets up the environment and initializes the command-line interface.
+func Initialize(ctx context.Context) {
+	// Set environment variables for the GCE project and DNS configurations.
+	_ = os.Setenv("ROACHPROD_DNS", "drt.crdb.io")
+	_ = os.Setenv("ROACHPROD_GCE_DNS_DOMAIN", "drt.crdb.io")
+	_ = os.Setenv("ROACHPROD_GCE_DNS_ZONE", "drt")
+	_ = os.Setenv("ROACHPROD_GCE_DEFAULT_PROJECT", "cockroach-drt")
+	_ = os.Setenv("ROACHPROD_GCE_DEFAULT_SERVICE_ACCOUNT", "622274581499-compute@developer.gserviceaccount.com")
+	// Initialize cloud providers for roachprod.
```
[the remark this answers is not in the document]
Comment (nameisbhaskar) [reply pkg/cmd/drtprod/cli/handlers.go:32]: This was hard-coded as drtprod commands were failing for me. I will remove this

**Review by vidit-bhat — APPROVED** [review]

**Comment by nameisbhaskar** [comment]
Thanks @sambhav-jain-16 @vidit-bhat for the review!

**Comment by nameisbhaskar** [comment]
bors r=@sambhav-jain-16,@vidit-bhat

**Comment by nameisbhaskar** [comment]
bors r=@sambhav-jain-16,@vidit-bhat

**Pull request merged**